src/assignments/tech-jobs-mvc.rst

```diff
@@ -3,4 +3,510 @@
+methods, with routes as annotated above their definitions. The first method
+renders a view that displays a table of clickable links for the different job
+categories. The second method renders a different view that displays
+information for the jobs that relate to a selected category. Both of the
+handlers obtain data by implementing the ``JobData`` class methods.
+
+In the ``listJobsByColumnAndValue`` method, the controller uses two query
+parameters passed in as ``column`` and ``value`` to determine what to fetch
+from ``JobData``. In the case of ``"all"`` it will fetch all job data,
+otherwise, it will retrieve a smaller set of information. The controller then
+renders the ``list-jobs.html`` view. We’ll explore the ``list.html`` and
+``list-jobs.html`` templates in a moment.
+
+``listJobsByColumnAndValue`` works similarly to the search functionality, in
+that we are "searching" for a particular value within a particular field and
+then displaying jobs that match. However, this is slightly different from the
+other way of searching in that the user will arrive at this handler method as a
+result of clicking on a link within the ``list`` view, rather than via
+submitting a form. We’ll see where these links originate when we look at the
+views. Also note that the ``listJobsByColumnAndValue`` method deals with an
+"all" scenario differently than if a user clicks one of the category links.
+
+The ``SearchController``
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Currently, the search controller contains only a single method, ``search``.
+It simply renders the form defined in the ``search.html`` template.
+
+Later in this assignment, you will receive instructions for adding a second
+handler to deal with user input and display the search results.
+
+The Views
+^^^^^^^^^
+
+Let’s turn our attention to the views.
+
+The ``fragments`` File
+~~~~~~~~~~~~~~~~~~~~~~~
+
+If the application is not running, launch it and navigate to the site’s home
+page in your browser. Also open up the
+``src/main/resources/templates/index.html`` file in IntelliJ. You’ll notice
+that there is a fair amount of content visible on the page that isn’t contained
+in ``index.html``. This is because we’re using two fragments from
+``fragments.html`` (``head`` and ``page-header``). These allow for some basic
+page structure and navigation to be shared across all of our views.
+
+Have a look at the structure of ``fragments.html``, but you will NOT need to do
+any work within this file for this assignment.
+
+.. admonition:: Tip
+
+   We use `Twitter’s Bootstrap <http://getbootstrap.com/>`__ CSS, HTML, and
+   JS framework to provide some styling and functionality to our views. The
+   appropriate files are included at the top of ``fragments.html`` and thus
+   are included on every page of our app.
+
+   You won’t have to explicitly use Bootstrap at all in this assignment,
+   but it’s a great way to make your sites look good with minimal work.
+   Consider using it on your own projects!
```

Comment:
"Consider using it in your own projects!"

Reply:
Fixed.  Thanks!